Makefile

```diff
@@ -35,7 +35,7 @@ infra pre ceph site main:
 requirements: requirements-python requirements-galaxy
 
 requirements-python: $(SELF)/requirements.txt
-	pip3 install --requirement $<
+	pip3 install --requirement $< 2> /dev/null || pip3 install --break-system-packages --requirement $<
```

Comment:
@sk4zuzu I think the confusion is that pip requires the --break-system-packages flag even when executed as non-root and installing the package locally to the user. It has no ability to break anything system-wide but it requires the flag nonetheless... I agree that the best way is using venvs though.

Reply:
Hi @1gramos

there is nothing that prevents users from executing this as root (with sudo for example). When you use this command as root, then new packages are installed in `/usr/local/lib/python3.11/dist-packages`  (example) , which can actually break something, there's a reason this message https://github.com/pypa/pip/blob/main/src/pip/_internal/exceptions.py#L680-L681 has been added.
Even when not installed system-wide it may break python for a specific non-root user :). 

We can't just casually throw in `--break-system-packages`, this is not our decision to make. I believe this flag is more intended to be used inside containers, where you can build and test some docker image. Here you can't test anything.

Users can execute this manually `pip3 install --break-system-packages --requirement requirements.txt` or consciously enable this workaround this in `~/.config/pip/pip.conf` or use environment variables (`PIP_BREAK_SYSTEM_PACKAGES=1`), we can state this in documentation for the time being and focus on implementing some venv based solution. There are plenty of scenarios we can make this work as you want, but not be responsible of breaking anything. :hugs: